src/approaches/grammar_search_invention_approach.py

```diff
@@ -49,44 +50,109 @@ def _generate(self) -> Iterator[Tuple[Predicate, float]]:
 
 
 @dataclass(frozen=True, eq=False, repr=False)
-class _SingleAttributeGEClassifier:
-    """Check whether a single attribute value on an object is >= some value.
+class _SingleAttributeCompareClassifier:
+    """Compare a single feature value with a constant value.
     """
     object_index: int
     attribute_name: str
-    value: float
+    constant: float
+    compare: Callable[[float, float], bool]
+    compare_str: str
 
     def __call__(self, s: State, o: Sequence[Object]) -> bool:
         obj = o[self.object_index]
-        return s.get(obj, self.attribute_name) >= self.value
+        return self.compare(s.get(obj, self.attribute_name), self.constant)
+
+    def __str__(self) -> str:
+        return f"({self.object_index}.{self.attribute_name}" + \
+               f"{self.compare_str}{self.constant:.3})"
 
 
 @dataclass(frozen=True, eq=False, repr=False)
 class _HoldingDummyPredicateGrammar(_PredicateGrammar):
     """A hardcoded cover-specific grammar.
+
     Good for testing with:
         python src/main.py --env cover --approach grammar_search_invention \
             --seed 0 --excluded_predicates Holding
     """
     def _generate(self) -> Iterator[Tuple[Predicate, float]]:
         # A necessary predicate
-        name = "InventedHolding"
```

Comment:
What happened to these names? What are the new names going to be? (just curious)

Reply:
(maybe include an example of the name of a generated predicate in the test file? how did the coverage check not complain...)

Reply:
the name is now `str(classifier)`. I'll add an example